flightlib/src/sensors/event_camera.cpp

```diff
@@ -0,0 +1,253 @@
+}
+bool EventCamera::setsigmaCp(const float sigma_cp) {
+  if (sigma_cp <= 0.0) {
+    logger_.warn(
+      "The setting value for Eventcamera sigmaCm is not valid, discard the "
+      "setting.");
+    return false;
+  }
+  sigma_cp_ = sigma_cp;
+  return true;
+}
+bool EventCamera::setsigmaCm(const float sigma_cm) {
+  if (sigma_cm <= 0.0) {
+    logger_.warn(
+      "The setting value for Eventcamera sigmaCm is not valid, discard the "
+      "setting.");
+    return false;
+  }
+  sigma_cm_ = sigma_cm;
+  return true;
+}
+bool EventCamera::setRefractory(const uint64_t refractory_period) {
+  if (refractory_period <= 0) {
+    logger_.warn(
+      "The setting value for Eventcamera refractory period is not valid, "
+      "discard the "
+      "setting.");
+    return false;
+  }
+  refractory_period_ns_ = refractory_period;
+  return true;
+}
+bool EventCamera::setLogEps(const float log_eps) {
+  if (log_eps <= 0.0) {
+    logger_.warn(
+      "The setting value for Eventcamera log_eps is not valid, discard the "
+      "setting.");
+    return false;
+  }
+  log_eps_ = log_eps;
+  return true;
+}
+bool EventCamera::setImgStore(const bool img_store) {
+  store_image_ = img_store;
+  return true;
+}
+
+bool EventCamera::changeTime(TimeMessage_t time_msg) {
+  if (time_msg.next_timestep <= 0) {
+    logger_.warn("timestep is zero or invalid");
+    return false;
+  }
+  sim_time = time_msg.current_time;
+  delta_time = time_msg.next_timestep;
+  return true;
+}
+double EventCamera::getSecSimTime() { return sim_time / 1000000.0; }
+int64_t EventCamera::getMicroSimTime() { return sim_time; }
+int64_t EventCamera::getNanoSimTime() { return sim_time * 1000; }
+int64_t EventCamera::getMicroTime() { return real_time; }
+
+Matrix<4, 4> EventCamera::getRelPose(void) const { return T_BC_; }
+
+int EventCamera::getWidth(void) const { return width_; }
+
+int EventCamera::getHeight(void) const { return height_; }
+
+Scalar EventCamera::getFOV(void) const { return fov_; }
+float EventCamera::getCm(void) const { return cm_; }
+float EventCamera::getCp(void) const { return cp_; }
+
+float EventCamera::getsigmaCm(void) const { return sigma_cm_; }
+float EventCamera::getsigmaCp(void) const { return sigma_cp_; }
+uint64_t EventCamera::getRefractory(void) const {
+  return refractory_period_ns_;
+}
+float EventCamera::getLogEps(void) const { return log_eps_; }
+bool EventCamera::getImgStore(void) { return store_image_; }
+
+bool EventCamera::getEventImages(cv::Mat& event) {
+  if (!event_image_queue_.empty()) {
+    // seems wrong here
+    event = event_image_queue_.front();
+    event_image_queue_.pop_front();
+    return true;
+  }
+  return false;
+}
+
+std::vector<Event_t> EventCamera::getEvents() {
+  std::vector<Event_t> events;
+  if (!event_queue_for_test.empty()) {
+    // seems wrong here
+    events = event_queue_for_test;
+    // event_queue_for_test.clear();
+    return events;
+  }
+  logger_.error("empty events buffer");
+  return events;
+}
```

Comment:
get which events? all events from the start or a single event at the current time step?